src/main/cpp/pop/Population.h

```diff
@@ -153,13 +153,26 @@ class Population
 		}
 		return total;
 	}
+  
+	/// Get the atlas.
+	const Atlas& getAtlas() const {return atlas;}
+
+	/// Register the map of GeoPositions to Towns to the atlas.
+	void AtlasRegisterTowns(Atlas::TownMap& towns) { atlas.RegisterTowns(towns); }
 
-	/// Store a GeoPosition in the population's atlas.
-	auto AtlasEmplace(const Atlas::Key& key, const geo::GeoPosition& pos) -> decltype(atlas.Emplace(key, pos))
+	/// Store a Cluster's GeoPosition in the population's atlas.
+	auto AtlasEmplaceCluster(const Atlas::ClusterKey& key, const geo::GeoPosition& pos)
+	    -> decltype(atlas.EmplaceCluster(key, pos))
 	{
-		return atlas.Emplace(key, pos);
+		return atlas.EmplaceCluster(key, pos);
+	}
+
+	Atlas::Town Hometown(const Person& person) const{
```

Comment:
If you change `atlas.LookupTown`'s return type to `const Atlas::Town&`, then you can have `Hometown` return a `const Atlas::Town&` as well (which should be slightly more efficient I guess).

Also, can you rename `Hometown` to something like `GetHometown` to make it easier to infer what this method does?